net: wifi: Changes needed to run Zephyr net samples with Wi-Fi (#11803)
nrfconnect/sdk-nrf: doc/nrf/samples/wifi_zephyr.rst

.. _wifi_zephyr_samples:

Wi-Fi: Zephyr networking samples
################################

> In addition to nRF Connect SDK samples, it is also possible to run selected Zephyr networking

Reviewer: should we mention "upstream Zephyr" samples here, to indicate that these are maintained in the upstream project?
e.g. "it is also possible to run selected TCP/IP networking samples provided and maintained as part of  the upstream Zephyr project"?

Author: I'll update this section, however I'd drop the `TCP/IP` part and stick to `networking samples`. IMO it's narrowing too much, we also support CoAP/DNS etc which are UDP.

Reviewer: That's fine. For me TCP/IP stack refers to the stack model name, not just the two protocols (i.e. it is another way to refer to the internet protocol suite)

samples with Wi-Fi.

> The following list specifies samples that are currently supported with the nRF700x driver:

Reviewer: same comment here.

Reviewer: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/wifi/index.html
(Wi-Fi driver is the name, at least in this page)

Author: I was inspired by "Working with nRF700x" section where it referred "nRF700x driver": https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/device_guides/working_with_nrf/nrf70/developing/constrained.html#id9
But I can switch to Wi-Fi driver if it's more commonly used.

Reviewer 2: Not sure if **nRF700X Wi-Fi driver** could be the alternative as used in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/drivers/nrf700x.html.

Reviewer: I am considering us phasing out the nRF700X term. Marketing has already switched to mostly used nRF70 Series, @richabp. Keep Wi-Fi driver for now, IMHO.

Reviewer 2: @ioannisg Sure!

* :ref:`dhcpv4-client-sample`
* :ref:`dns-resolve-sample`
* :ref:`ipv4-autoconf-sample`
* :ref:`mdns-responder-sample`
* :ref:`mqtt-publisher-sample`
* :ref:`mqtt-sn-publisher-sample`
* :ref:`coap-client-sample`
* :ref:`coap-server-sample`
* :ref:`sockets-echo-sample`
* :ref:`async-sockets-echo-sample`
* :ref:`sockets-echo-client-sample`
* :ref:`sockets-echo-server-sample`
* :ref:`sockets-http-get`
* :ref:`sntp-client-sample`
* :ref:`syslog-net-sample`
* :ref:`telnet-console-sample`

> Building and running

Author: I've added a Configuration section with details on how to configure Wi-Fi credentials.

The Dependencies section IMO doesn't make much sense, as we're not documenting a single example here, each sample has its own dependencies.

As for requirements, I'm a bit puzzled here. We typically use `.. table-from-sample-yaml::` for this, but here that's not the case, as there's is no corresponding `sample.yaml` with this documentation.

********************

All of the listed samples have been provided with an overlay file, ``overlay-nrf700x.conf``, which

> allows to configure the sample to run with nRF700x driver.

Reviewer: Is "nRF700x driver" the right name? IIRC, we use "Wi-Fi driver" in NCS, but up to finding a better name if needed.

> All of the listed samples have been provided with an overlay file, ``overlay-nrf700x.conf``, which
> allows to configure the sample to run with Wi-Fi driver.

Reviewer 2: ```suggestion
An overlay file, ``overlay-nrf700x.conf`` is provided to all Zephyr samples, which configures the sample to run with the Wi-Fi driver.
```

To build Zephyr samples for the nRF7002 DK, use the ``nrf7002dk_nrf5340_cpuapp`` build target.
The following is an example of the CLI command:

.. code-block:: console

   west build -b nrf7002dk_nrf5340_cpuapp -- -DOVERLAY_CONFIG=overlay-nrf700x.conf

> To build for the nRF7002 EK with nRF5340 DK, use the ``nrf5340dk_nrf5340_cpuapp`` build target with
> the ``SHIELD`` CMake option set to ``nrf7002ek``.

Reviewer 2: ```suggestion
To build for the nRF7002 EK with nRF5340 DK, use the ``nrf5340dk_nrf5340_cpuapp`` build target with the ``SHIELD`` CMake option set to ``nrf7002ek``.
```
